libs/sdl/sdl.c

```diff
@@ -904,66 +928,71 @@ HL_PRIM char* HL_NAME(get_clipboard_text)() {
 }
 
 HL_PRIM void HL_NAME(set_drag_and_drop_enabled)( bool enabled ) {
-	SDL_EventState(SDL_DROPFILE, enabled ? SDL_ENABLE : SDL_DISABLE);
+	SDL_SetEventEnabled(SDL_EVENT_DROP_FILE, enabled );
 }
 
 HL_PRIM bool HL_NAME(get_drag_and_drop_enabled)() {
-	return SDL_EventState(SDL_DROPFILE, SDL_QUERY);
+	return SDL_EventEnabled(SDL_EVENT_DROP_FILE);
 }
 
 HL_PRIM varray* HL_NAME(get_displays)() {
-	int n = SDL_GetNumVideoDisplays();
-	if (n < 0)
+	int n;
+	SDL_DisplayID *displays = SDL_GetDisplays(&n);
+	if (n <= 0)
 		return NULL;
 	varray* arr = hl_alloc_array(&hlt_dynobj, n);
 	for (int i = 0; i < n; i++) {
 		vdynamic *obj = (vdynamic*) hl_alloc_dynobj();
 		SDL_Rect rect;
-		SDL_GetDisplayBounds(i, &rect);
+		SDL_DisplayID display = displays[i];
+		SDL_GetDisplayBounds(display, &rect);
 		hl_dyn_seti(obj, hl_hash_utf8("right"), &hlt_i32, rect.x+rect.w);
 		hl_dyn_seti(obj, hl_hash_utf8("bottom"), &hlt_i32, rect.y+rect.h);
 		hl_dyn_seti(obj, hl_hash_utf8("left"), &hlt_i32, rect.x);
 		hl_dyn_seti(obj, hl_hash_utf8("top"), &hlt_i32, rect.y);
 		hl_dyn_seti(obj, hl_hash_utf8("handle"), &hlt_i32, i);
-		const char *name = SDL_GetDisplayName(i);
+		const char *name = SDL_GetDisplayName(display);
 		hl_dyn_setp(obj, hl_hash_utf8("name"), &hlt_bytes, hl_copy_bytes(name, (int) strlen(name)+1));
 		hl_aptr(arr, vdynamic*)[i] = obj;
 	}
 	return arr;
 }
 
 HL_PRIM varray* HL_NAME(get_display_modes)(int display_id) {
-	int n = SDL_GetNumDisplayModes(display_id);
-	if (n < 0)
-		return NULL;
-	varray* arr = hl_alloc_array(&hlt_dynobj, n);
-	for (int i = 0; i < n; i++) {
-		SDL_DisplayMode mode;
-		SDL_GetDisplayMode(display_id, i, &mode);
+
+
+	SDL_DisplayMode **modes;
+	int count;
+	modes = SDL_GetFullscreenDisplayModes(display_id, &count);

```

Comment:
Calling `SDL_GetFullscreenDisplayModes` without freeing the result leaks memory.